worker/src/handlers/schedule.ts

```diff
@@ -1,181 +1,208 @@
 // Scheduled taks handler to manage the KV store
 import { CurrentAnalytics } from "../../../site/src/data";
-import { SanitizedPayload } from "../data";
+import { Queue, QueueData, SanitizedPayload } from "../data";
 import { average } from "../utils/average";
 import { formatDate } from "../utils/date";
 
 export async function handleSchedule(event: ScheduledEvent): Promise<void> {
-  const core_analytics: Record<string, any> = {};
-  const storedAnalytics =
-    (await KV.get<{ [key: string]: CurrentAnalytics }>(
-      "core_analytics",
-      "json"
-    )) || {};
-  const storedData = await listStoredData();
-  console.log(JSON.stringify(storedData));
-  //const currentDataset: CurrentAnalytics = generateCurrentDataset(storedData);
-
-  const currentDate = formatDate(new Date());
-  const currentDateObj = new Date(
-    currentDate.year,
-    currentDate.month,
-    currentDate.day,
-    currentDate.hour
-  );
-  const timestampString = String(currentDateObj.getTime());
-  //const storeKey = `history:${timestampString}`;
-
-  const dataKeys = Object.keys(storedAnalytics);
-  const lastDataEntry = storedAnalytics[dataKeys[dataKeys.length - 1]];
-
-  for (const key of dataKeys) {
-    core_analytics[key] = {
-      active_installations: storedAnalytics[key].active_installations,
-      installation_types: storedAnalytics[key].installation_types,
-    };
-  }
-
-  core_analytics[timestampString] = {
-    ...lastDataEntry,
-    active_installations: storedData,
+  await prosessQueue();
+}
+
+async function prosessQueue(): Promise<void> {
+  const queue = (await KV.get<Queue>("queue", "json")) || {
+    entries: [],
+    data: {},
   };
 
-  //await KV.put(storeKey, JSON.stringify(currentDataset));
-  await KV.put("core_analytics", JSON.stringify(core_analytics));
+  if (queue.entries.length === 0) {
+    // No entries, get list
+    console.log("No entries, get list");
+    queue.entries = await listKV("uuid");
+  }
+
+  // Prosess the first 850 entries in the array
+  for (const entryKey of queue.entries.slice(0, 850)) {
+    let entryData;
+    try {
+      entryData = await KV.get<SanitizedPayload>(entryKey, "json");
+    } catch (e) {
+      console.log(e);
+    }
+
+    if (entryData !== undefined && entryData !== null) {
+      queue.data = combineEntryData(queue.data, entryData);
+    }
+  }
+  queue.entries = queue.entries.slice(850);
+
+  if (queue.entries.length === 0) {
+    // No more entries, store and reset queue data
+    console.log("No more entries, store and reset queue data");
+    const core_analytics: Record<string, any> = {};
+    const timestampString = new Date().getTime();
+
+    const queue_data = prosessQueueData(queue.data as QueueData);
```

Comment:
Why is the `as` necessary?

```suggestion
    const queue_data = processQueueData(queue.data as QueueData);
```

Reply:
because `data: {}` does not match the type of `QueueData`